sqldb/sqlc/queries/amp_invoices.sql

```diff
@@ -1,79 +1,67 @@
--- name: InsertAMPInvoicePayment :exec
-INSERT INTO amp_invoice_payments (
-    set_id, state, created_at, settled_index, invoice_id
+-- name: UpsertAmpInvoice :execresult
+INSERT INTO amp_invoices (
+    set_id, state, created_at, invoice_id
 ) VALUES (
-    $1, $2, $3, $4, $5
-); 
+    $1, $2, $3, $4
+) ON CONFLICT (set_id, invoice_id) DO NOTHING;
 
--- name: SelectAMPInvoicePayments :many
-SELECT aip.*, ip.*
-FROM amp_invoice_payments aip LEFT JOIN invoice_payments ip ON aip.settled_index = ip.id
-WHERE (
-    set_id = sqlc.narg('set_id') OR 
-    sqlc.narg('set_id') IS NULL
-) AND (
-    aip.settled_index = sqlc.narg('settled_index') OR 
-    sqlc.narg('settled_index') IS NULL
-) AND (
-    aip.invoice_id = sqlc.narg('invoice_id') OR 
-    sqlc.narg('invoice_id') IS NULL
-); 
+-- name: GetAmpInvoiceID :one
+SELECT invoice_id FROM amp_invoices WHERE set_id = $1;
 
--- name: UpdateAMPPayment :exec
-UPDATE amp_invoice_payments
-SET state = $1, settled_index = $2
-WHERE state = 0 AND (
-    set_id = sqlc.narg('set_id') OR 
-    sqlc.narg('set_id') IS NULL
-) AND (
-    invoice_id = sqlc.narg('invoice_id') OR 
-    sqlc.narg('invoice_id') IS NULL
-); 
+-- name: UpdateAmpInvoiceState :exec
+UPDATE amp_invoices
+SET state = $2, 
+    settled_index = COALESCE(settled_index, $3),
+    settled_at = COALESCE(settled_at, $4)
+WHERE set_id = $1; 
 
--- name: InsertAMPInvoiceHTLC :exec
+-- name: InsertAmpInvoiceHtlc :exec
 INSERT INTO amp_invoice_htlcs (
-    set_id, htlc_id, root_share, child_index, hash, preimage
+    invoice_id, set_id, htlc_id, root_share, child_index, hash, preimage
 ) VALUES (
-    $1, $2, $3, $4, $5, $6
+    $1, $2, $3, $4, $5, $6, $7
 );
 
--- name: GetAMPInvoiceHTLCsBySetID :many
-SELECT *
-FROM amp_invoice_htlcs
-WHERE set_id = $1;
-
--- name: GetAMPInvoiceHTLCsByInvoiceID :many
+-- name: FetchAMPInvoices :many
 SELECT *
-FROM amp_invoice_htlcs
-WHERE invoice_id = $1;
-
--- name: GetSetIDHTLCsCustomRecords :many
-SELECT ihcr.htlc_id, key, value
-FROM amp_invoice_htlcs aih JOIN invoice_htlc_custom_records ihcr ON aih.id=ihcr.htlc_id 
-WHERE aih.set_id = $1;
-
--- name: UpdateAMPInvoiceHTLC :exec
-UPDATE amp_invoice_htlcs
-SET preimage = $1
-WHERE htlc_id = $2;
-
--- name: DeleteAMPHTLCCustomRecords :exec
-WITH htlc_ids AS (
-    SELECT htlc_id
-    FROM amp_invoice_htlcs 
-    WHERE invoice_id = $1
-)
-DELETE
-FROM invoice_htlc_custom_records
-WHERE htlc_id IN (SELECT id FROM htlc_ids);
+FROM amp_invoices
+WHERE invoice_id = $1 
```

Comment:
For many queries (at least with the way things work today), we also want to be able to query by `payment_addr` as well. 

IMO we only care about fetching by `set_id` for the set of _HTLCs_. For a given AMP invoice with a `payment_addr`, there'll only be a single invoice in the DB, but many HTLCs with diff set IDs linking to it. A new `set_id` is generated each time a new payment is sent to the invoice.